### PR TITLE
F creacion edicion members ot105 35

### PR DESCRIPTION
**Pull request by pdvallejos**
Summary:
se agrega ruta y componente para creacion y edicion de miembros.

Link: https://alkemy-labs.atlassian.net/browse/OT105-35

Changes added:
se agrega ruta el archivo app mencionada anteriormente
se agrega archivo MembersCreateEdit.js a la carpeta Members

image

### CODE REVIEW

**Comment by saralavanchy** [comment]
Please clean the title to be clear

File: src/Components/Members/MembersCreateEdit.js [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+  height: 60,
+  padding: 4,
+  boxSizing: 'border-box',
+};
+
+function MembersCreateEdit() {
+  const { id } = useParams();
+  const urlEdit = `http://ongapi.alkemy.org/public/api/members/${id}`;
+  const urlCreate = `http://ongapi.alkemy.org/public/api/members`;
+  const [initialValues, setInitialValues] = useState({
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js]: please use a mor specif name, initialValues is too generic and do not give information about its content. Maybe member is a more specific name

File: src/Components/Members/MembersCreateEdit.js [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+const thumb = {
+  display: 'inline-flex',
+  borderRadius: 2,
+  border: '1px solid #eaeaea',
+  margin: 'auto',
+  width: 'auto',
+  height: 60,
+  padding: 4,
+  boxSizing: 'border-box',
+};
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js]: please move this style to Styles folder

File: src/Components/Members/MembersCreateEdit.js [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+function MembersCreateEdit() {
+  const { id } = useParams();
+  const urlEdit = `http://ongapi.alkemy.org/public/api/members/${id}`;
+  const urlCreate = `http://ongapi.alkemy.org/public/api/members`;
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js]: this both Url should be here, they should be on memeberService

File: src/Components/Members/MembersCreateEdit.js [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+  const [initialValues, setInitialValues] = useState({
+    name: '',
+    image: '',
+    description: '',
+    facebookUrl: '',
+    linkedinUrl: '',
+  });
+
+  const [files, setFiles] = useState([]);
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js]: what this files are for? files are a too generic name. Please use one more specific

Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js]: what this files are for? files are a too generic name. Please use one more specific

File: src/Components/Members/MembersCreateEdit.js [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+    accept: 'image/png, image/jpg',
+    onDrop: (acceptedFiles) => {
+      setInitialValues({
+        ...initialValues,
+        image: acceptedFiles.map((file) =>
+          Object.assign(file, {
+            preview: URL.createObjectURL(file),
+          }),
+        ),
+      });
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js:39]: what is being done inside of setInitialValues, image? The answer of that question is a good candidate for the name of the function where this peace of code should be. Please move this to a separated function and then call it here.
```
const getImageWithUrl= (acceptedFiles) => acceptedFiles.map((file) =>
          Object.assign(file, {
            preview: URL.createObjectURL(file),
          }),
        );
```
then on the onFileDrop fuction
```
setInitialValues({...initialValue, image: getFilesWithUrl(acceptedFiles)})
```

File: src/Components/Members/MembersCreateEdit.js [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+      setFiles(
+        acceptedFiles.map((file) =>
+          Object.assign(file, {
+            preview: URL.createObjectURL(file),
+          }),
+        ),
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js:45]: same about getImageWithUrl

File: src/Components/Members/MembersCreateEdit.js [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+      );
+    },
+  });
+  const thumbs = files.map((file) => (
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js]: please use a more specific name

File: src/Components/Members/MembersCreateEdit.js [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+    <div key={file.name} style={thumb}>
+      <div className="thumb-inner">
+        <img className="img" src={file.preview} />
+      </div>
+    </div>
+  ));
+
+  useEffect(
+    () => () => {
+      // Make sure to revoke the data uris to avoid memory leaks
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js]: please delete this comment, the code should be enought documentation for itself

File: src/Components/Members/MembersCreateEdit.js [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+      files.forEach((file) => URL.revokeObjectURL(file.preview));
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js]: move this to a function outside the useEffect
```
const avoidMemoryLeak = () =>  files.forEach((file) => URL.revokeObjectURL(file.preview));
```

File: src/Components/Members/MembersCreateEdit.js [hunk 10]
```diff
@@ -0,0 +1,316 @@
...
+    // eslint-disable-next-line no-console
+    console.log(values);
+    setInitialValues({ ...initialValues, name: e.target.value });
+  };
+  const facebookChange = (e) => {
+    setInitialValues({ ...initialValues, facebookUrl: e.target.value });
+  };
+  const linkedInChange = (e) => {
+    setInitialValues({ ...initialValues, linkedinUrl: e.target.value });
+  };
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js:84]: for name, facebook and linkedin you should create only one onChange and use a switch on it

File: src/Components/Members/MembersCreateEdit.js [hunk 11]
```diff
@@ -0,0 +1,316 @@
...
+  const handleClick = () => {
+    const datosConEtiquetas = initialValues.description;
+    const datosSinEtiquetas = datosConEtiquetas.replace(/<[^>]+>/g, '');
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js]: move the replace into a function "cleanTags"

File: src/Components/Members/MembersCreateEdit.js [hunk 12]
```diff
@@ -0,0 +1,316 @@
...
+        .then((response) => {
+          if (response.status === 200) {
+            // eslint-disable-next-line no-console
+            alert('Edit member successfully');
+
+            return setInitialValues(initialValues);
+          }
+        })
+        .catch((e) => alert(e));
+    }
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js]: all this should be on MembersService

File: src/Components/Members/MembersCreateEdit.js [hunk 13]
```diff
@@ -0,0 +1,316 @@
...
+        if (name) setInitialValues({ ...initialValues, name: name });
+        if (image) setInitialValues({ ...initialValues, image: image });
+        if (description)
+          setInitialValues({ ...initialValues, description: description });
+        if (facebookUrl)
+          setInitialValues({ ...initialValues, facebookUrl: facebookUrl });
+        if (linkedinUrl)
+          setInitialValues({ ...initialValues, linkedinUrl: linkedinUrl });
+      }
+    };
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js:94]: please move this to a function outside the useEffect. Also dataEdit do not describe what is being done here. Please update the name to something like loadMemberInfoOnEdit

File: src/Components/Members/MembersCreateEdit.js [hunk 14]
```diff
@@ -0,0 +1,316 @@
...
+        name: '',
+        image: '',
+        description: '',
+        facebookUrl: '',
+        linkedinUrl: '',
+      }}
+      validate={(values) => {
+        let errors = {};
+
+        //Name
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js]: The code should be enought documentation to itself, please delete this comment

File: src/Components/Members/MembersCreateEdit.js [hunk 15]
```diff
@@ -0,0 +1,316 @@
...
+        if (!values.name) {
+          errors.name = 'Please submit a email';
+        }
+        //Image
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js]: The code should be enought documentation to itself, please delete this comment

File: src/Components/Members/MembersCreateEdit.js [hunk 16]
```diff
@@ -0,0 +1,316 @@
...
+        if (values.image === null) {
+          errors.image = 'Please submit a image';
+        } else if (values.image) {
+          errors.image = false;
+        }
+        //Description
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js]: The code should be enought documentation to itself, please delete this comment

File: src/Components/Members/MembersCreateEdit.js [hunk 17]
```diff
@@ -0,0 +1,316 @@
...
+                error={Boolean(touched.name && errors.name)}
+                id="name"
+                label="Name"
+                name="name"
+                placeholder="complet name"
+                type="text"
+                value={initialValues.name}
+                onChange={(e, values) => nameChange(e, values)}
+              />
+              <ErrorMessage
```
Comment (saralavanchy) [inline src/Components/Members/MembersCreateEdit.js]: please create a function to show Errors
```
const showErrors = (attribute) => (<ErrorMessage
                component={() => (
                  <Alert severity="warning">{errors[attribute]}</Alert>
                )}
                name="name"
              />)
```

File: src/Components/News/NewsCreateEdit.js [hunk 18]
```diff
@@ -0,0 +1,247 @@
+//eslint-disable react-hooks/exhaustive-deps
```
Comment (saralavanchy) [inline src/Components/News/NewsCreateEdit.js:1]: This file is not scope of this user story

**Review by saralavanchy — CHANGES_REQUESTED** [review]

File: src/Components/Members/MembersCreateEdit.js (hunk 5 above)
Comment (pdvallejos) [reply src/Components/Members/MembersCreateEdit.js:39]: Sara de la manera que lo he cambiado  para tratar de sacarlo siempre me tira error o en los map o sino me tira este error
image

**Review by saralavanchy — APPROVED** [review]

**Pull request merged**